esp/tools/soapplus/http.cpp

```diff
@@ -1229,40 +1268,130 @@ int HttpClient::sendStressRequest(StringBuffer& request, HttpStat* stat)
 
     unsigned start = msTick();
 
-    Owned<CSimpleSocket> sock = new CSimpleSocket(m_ssctx.get(), m_logfile);
-    
-    int ret = sock->connect(m_serveraddr.get());
-    if(ret < 0)
+    Owned<CSimpleSocket> sock;
+    bool isPersistent = m_globals->getPropBool("isPersist", false);
+    if (isPersistent)
     {
-        return -1;
+        if (persistentSocket.get() == nullptr)
+            persistentSocket.setown(new CSimpleSocket(m_ssctx.get(), m_logfile));
+        sock.set(persistentSocket.get());
+    }
+    else
+        sock.setown(new CSimpleSocket(m_ssctx.get(), m_logfile));
+    if (!sock->isConnected())
+    {
+        int ret = sock->connect(m_serveraddr.get());
+        if (ret < 0)
+        {
+            return -1;
+        }
     }
 
+    bool shouldClose = false;
     int sent = sock->send(request);
     __int64 total_len = 0;
-    if(sent >= 0)
+    StringBuffer xml;
+    if (sent > 0)
     {
-        int len = 1;
+        int len = 0;
         char recvbuf[2048];
-        StringBuffer xml;
-        while(1)
+        if (!isPersistent)
+        {
+            while (1)
+            {
+                len = sock->receive(recvbuf, 2047);
+                if (len > 0)
+                {
+                    total_len += len;
+                    recvbuf[len] = 0;
+                    if (m_doValidation)
+                        xml.append(len, recvbuf);
+                    if (http_tracelevel >= 10)
+                        fprintf(m_logfile, "%s", recvbuf);
+                }
+                else
+                    break;
+            }
+        }
+        else
         {
-            len = sock->receive(recvbuf, 2047);
-            if(len > 0)
+            StringBuffer headersbuf;
+            unsigned int searchStart = 0;
+            while (1)
             {
-                total_len += len;
-                recvbuf[len] = 0;
-                if (m_doValidation)
-                    xml.append(len, recvbuf);
-                if(http_tracelevel >= 10)
-                    fprintf(m_logfile, "%s", recvbuf);
+                len = sock->receive(recvbuf, 0, 2047);
```

Comment:
Should min be 0 here ?  Maybe does not matter but not sure.

Reply:
This shouldn't matter. Socket will block until either data arrives or connection is closed or error happens, in another word, there's no case where it just receives 0 byte without connection close or error. Hard to tell the minimum header length, but 0 is safe here.

Reply:
@mckellyln I looked at the code further. 0 is fine unless EINTR happens, so the min is better to be 1 here.

Reply:
Actually the ENITR issue can be avoided by adding a "continue" there. But I'll still change 0 to 1 as I do expect some data (at least 1). I'll also add the continue, as the rest of the loop is unnecessary for that case.